criu/net.c

```diff
@@ -2437,12 +2426,38 @@ static inline int restore_nftables(int pid)
 	nft_ctx_free(nft);
 buf_free_out:
 	xfree(buf);
+out:
+	return ret;
+}
+#endif
+
+static inline int restore_nftables(int pid)
+{
+	int ret = -1;
+	struct cr_img *img;
+
+	img = open_image(CR_FD_NFTABLES, O_RSTR, pid);
+	if (img == NULL)
+		return -1;
+	if (empty_image(img)) {
+		/* Backward compatibility */
+		pr_info("Skipping nft restore, no image\n");
+		ret = 0;
+		goto image_close_out;
+	}
+
+#if defined(CONFIG_HAS_NFTABLES_LIB_API_0) || defined(CONFIG_HAS_NFTABLES_LIB_API_1)
+	ret = do_restore_nftables(img);
```

Comment:
Returning the return value of nested function is a bad practice, this way understanding allowed return values range of the function becomes harder, as you need to investigate nested functions recursively. I would advise:
```
if (!do_restore_nftables(img))
        exit_code = 0;
```
Meaning that any bad return from do_restore_nftables converts to -1 explicitly.

Reply:
Thanks! Have done.